FV3/coarse_graining/coarse_graining.F90

```diff
@@ -798,13 +827,35 @@ subroutine remap_edges_along_x(field, phalf, dx, ptop, result)
     enddo
 
     ! 5. Create mask
-    do k = 1, npz
-      where (coarse_phalf_d_grid_on_fine(:,:,k+1) .lt. phalf_d_grid(:,:,npz+1))
-        mask(:,:,k) = .true.
-      elsewhere
-        mask(:,:,k) = .false.
-      endwhere
-    enddo
+    if (extrapolate) then
+      allocate(coarse_pfull_d_grid_on_fine(is:ie,js_coarse:je_coarse+1,1:npz))
+      call compute_pfull_from_phalf_d_grid_y(coarse_phalf_d_grid_on_fine, coarse_pfull_d_grid_on_fine)
```

Comment:
I suspect this typo is why coarse-graining with this mode does not reproduce across layouts for `fv_core.res` restart files.  Will push a fix after confirming this locally.
```suggestion
      call compute_pfull_from_phalf_d_grid_x(coarse_phalf_d_grid_on_fine, coarse_pfull_d_grid_on_fine)
```